GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/DownloadObjectOptions.cs

```diff
@@ -25,5 +31,26 @@ internal void ModifyDownloader(MediaDownloader downloader)
                 downloader.ChunkSize = ChunkSize.Value;
             }
         }
+
+        /// <summary>
+        /// Appends any parameters to the given builder, which is guaranteed to
+        /// already have one query parameter (alt=media).
+        /// </summary>
+        internal void AppendUriParameters(StringBuilder uriBuilder)
+        {
+            MaybeAppendParameter(uriBuilder, "generation", Generation);
+        }
+
+        private static void MaybeAppendParameter(StringBuilder uriBuilder, string name, long? value)
+        {
+            if (value != null)
+            {
+                uriBuilder
+                    .Append("&")
```

Comment:
It feels a bit redundant parsing the URL to check there's a query parameter, when we know this is only used by AppendUriParameters which is documented to need a query parameter already... I think we're okay. If this were used more generally (e.g. by other operations) or if it were public, that would be a different matter.